main/modes/mode_diceroller.c

```diff
@@ -0,0 +1,635 @@
+#include <stdio.h>
+#include <stdint.h>
+#include <stddef.h>
+#include <string.h>
+#include <stdlib.h>
+#include <malloc.h>
+#include <math.h>
+
+#include "cndraw.h"
+#include "esp_timer.h"
+#include "swadgeMode.h"
+#include "swadge_esp32.h"
+#include "led_util.h" // leds
+#include "meleeMenu.h"
+#include "mode_main_menu.h"
+#include "nvs_manager.h"
+#include "bresenham.h"
+#include "display.h"
+#include "embeddednf.h"
+#include "embeddedout.h"
+#include "musical_buzzer.h"
+#include "settingsManager.h"
+#include "linked_list.h"
+//#include "bootloader_random.h"
+#include "esp_random.h"
+
+#include "swadge_util.h"
+#include "text_entry.h"
+
+#include "mode_diceroller.h"
+
+
+//TODO: Add Doxygen Blocks
+//TODO: Add Cosmetic Features of Colored background/foreground objects. Add Smoother animations. Add LED interaction.
+
+const double pi = 3.14159265359;
+
+void diceEnterMode(display_t* disp);
+void diceExitMode(void);
+void diceMainLoop(int64_t elapsedUs);
+void diceButtonCb(buttonEvt_t* evt);
+//void diceTouchCb(touch_event_t* evt);
+//void diceAccelerometerCb(accel_t* accel);
+//void diceAudioCb(uint16_t* samples, uint32_t sampleCnt);
+coord_t* getRegularPolygonVertices(int8_t sides, float rotDeg, int16_t radius);
+void drawRegularPolygon(int xCenter, int yCenter, int8_t sides, float rotDeg, int16_t radius, paletteColor_t col, int dashWidth);
+void changeActiveSelection();
+void changeDiceCountRequest(int change);
+void changeDiceSidesRequest(int change);
+void doRoll(int count, int sides, int index);
+void doStateMachine(int64_t elapsedUs);
+
+const int MAXDICE = 6;
+const int COUNTCOUNT = 7;
+int validSides[] = {4, 6, 8, 10, 12, 20, 100};
+int polygonSides[] = {3, 4, 3, 4, 5, 3, 6};
+
+int64_t rollAnimationPeriod = 1000000; //1 Second Spin

```

Comment:
Try to stick to `int32_t` unless really necessary.

Reply:
limited int64_t usage to esp_clock timestamps